src/main/java/com/uid2/client/IdentityMapV3Response.java

```diff
@@ -0,0 +1,128 @@
+package com.uid2.client;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IdentityMapV3Response {
+    IdentityMapV3Response(String response, IdentityMapV3Input identityMapInput) {
+        ApiResponse apiResponse = new Gson().fromJson(response, ApiResponse.class);
+        status = apiResponse.status;
+
+        if (!isSuccess()) {
+            throw new Uid2Exception("Got unexpected identity map status: " + status);
+        }
+
+        populateIdentities(apiResponse.body, identityMapInput);
+    }
+
+    private void populateIdentities(Map<String, List<ApiIdentity>> apiResponse, IdentityMapV3Input identityMapInput) {
+        for (Map.Entry<String, List<ApiIdentity>> identitiesForType : apiResponse.entrySet()) {
+            populateIdentitiesForType(identityMapInput, identitiesForType.getKey(), identitiesForType.getValue());
+        }
+    }
+
+    private void populateIdentitiesForType(IdentityMapV3Input identityMapInput, String identityType, List<ApiIdentity> identities) {
+        for (int i = 0; i < identities.size(); i++) {
+            ApiIdentity apiIdentity = identities.get(i);
+            List<String> rawDiis = identityMapInput.getRawDiis(identityType, i);
```

Comment:
Maybe inputDii is a better name?

Reply:
Sure, the old version used rawDiis but happy to rename it for clarity.

Reply:
it might mean raw as opposed to hashed and normalized DII - could you please check?